src/rpc/state_api.rs

```diff
@@ -377,3 +377,27 @@ pub(in crate::rpc) async fn state_get_randomness_from_beacon<
     )?;
     Ok(LotusJson(value.to_vec()))
 }
+
+/// Get read state
+pub(in crate::rpc) async fn state_read_state<DB: Blockstore + Send + Sync + 'static>(
+    data: Data<RPCState<DB>>,
+    Params(LotusJson((addr, tsk))): Params<LotusJson<(Address, TipsetKeys)>>,
+) -> Result<LotusJson<ApiActorState>, JsonRpcError> {
+    let ts = data.chain_store.load_required_tipset(&tsk)?;
+    let actor = data
+        .state_manager
+        .get_actor(&addr, *ts.parent_state())?
+        .ok_or("Actor address could not be resolved")?;
+    let blk = data
+        .state_manager
+        .blockstore()
+        .get(&actor.state)?
+        .ok_or("Failed to get block from blockstore")?;
+    let state = fvm_ipld_encoding::from_slice::<Vec<Cid>>(&blk)?[0];
```

Comment:
What happens if the vector of `Cids` is empty? Is it even possible?

Reply:
Good question. Let's add a test for that.

Reply:
added